scripts/smartcommit/done.js

```diff
@@ -0,0 +1,128 @@
+//# sourceURL=done.js
+'use strict';
+
+//  Description
+//
+// This script work as a handler that sets version when transitioning to done.
+// Version should be present in version input argument.
+// Command should look like #done version=2.0.
+// Please take into account that resolution of issue remains empty after transition.
+// https://github.com/BigBrassBand/jira-git-workflow-hooks
+//
+//  Applying
+// Set appropriate keys for error messages in constants: NO_STATUS, NO_ALLOWED_ACTIONS_TEMPLATE, UNKNOWN_VERSION.
+// Command name will be used as transition name.
+//
+//  Classes passed as parameters by default
+//
+//  The passed smart commit command
+// command       : com.atlassian.jira.plugins.dvcs.smartcommits.model.CommitCommands.CommitCommand
+//
+//  The committer user
+// user          : com.atlassian.jira.user.ApplicationUser
+//
+//  The issue specified in smart commit
+// issue         : com.atlassian.jira.issue.MutableIssue
+//
+//  A command result to return execution errors
+// commandResult : com.atlassian.jira.plugins.dvcs.smartcommits.model.CommandsResults.CommandResult
+//
+//  The commit revision ID
+// revision : java.lang.String
+//
+//  The full commit message
+// commitComment : java.lang.String
+//
+//  The commit timestamp (unixtime format)
+// commitTime : long
+//
+//  An e-mail of the commit author
+// authorEmail : java.lang.String
+//
+//  An author name of the commit
+// authorName : java.lang.String
+(function () {
+    load(__DIR__ + '../utils.js');
+
+    //get version param from input params
+    function getVersionParam() {
+        if (command.getArguments().length > 0 && command.getArguments().get(0).trim().length > 0) {
+            var regexp = /version=((\d|.)+)/;
```

Comment:
Regexp looks wrong. It will accept versions like "version=some cool version". Do you think it is correct?
This regexp is equal to the next one:  "/version=(.+)/" -- i.e. it accepts any characters after "version=".

I think it will be better to use something like the next:  "/version=([\w.-]+)/"
Please test if it accepts all possible versions correctly.

Reply:
I will correct it